### PR TITLE
76 ui stats battles

### LINKED ISSUE #58: Create player factory
Create a factory for creating units on scene start

### LINKED ISSUE #75: ORM for Database
Create a way to access the database information for C# and Unity easily

### LINKED ISSUE #76: Stats UI in battle
Optional canvas that displays a Units stats

### PR DESCRIPTION
**Pull request by Toeecko**
Fixes #58 
Fixes #75 
Fixes #76 

Changes proposed in this pull request:
- Creates character factory
- Creates db
- Creates ORM
- Creates StatsUI

### CODE REVIEW

**Review by ThBrokenToaster — DISMISSED** [review]

File: Assets/Scripts/Battlefield/CreatureScripts/UnitStats.cs [hunk 1]
```diff
@@ -9,12 +9,14 @@ public class UnitStats : MonoBehaviour
...
+        public int magicAttack;
         public int defense;
-        public int specialDefense;
+        public int magicDefense;
         public int speedIntit;
         public int movement;
-
+        public int accuracy;
+        public int evasion;
+        public string role;
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/CreatureScripts/UnitStats.cs:19]: This file should be removed. We already have an interface and class for this and this was not part of these issues tasking. I am willing to accept this PR with these changes as long as there is a high priority issue created for next timebox focused on removing this

File: Assets/Scripts/Battlefield/Camera/CameraManager.cs [hunk 2]
```diff
@@ -11,12 +11,12 @@ public class CameraManager : MonoBehaviour
     {
 
         public TurnManager turnManager;
-        public GameObject[] cameras = new GameObject[5];
+        public List<GameObject> cameras = new List<GameObject>();
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/Camera/CameraManager.cs:14]: Why was this changed? This seems to serve no purpose

File: Assets/Scripts/Battlefield/TurnMechanism/TurnManager.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace SwordAndBored.Battlefield.TurnMechanism
     public class TurnManager : MonoBehaviour
     {
         [Header("Units")]
-        public GameObject[] units;
+        public List<GameObject> units = new List<GameObject>();
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/TurnMechanism/TurnManager.cs:14]: Why was this changed? This seems to serve no purpose

File: Assets/Scripts/Battlefield/CreatureScripts/UnitAbilitiesContainer.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@ namespace SwordAndBored.Battlefield.CreaturScripts
     public class UnitAbilitiesContainer : MonoBehaviour
     {
         [Header("Ability Info")]
-        public AbstractAbilitySO[] abilities;
+        public List<AbstractAbilitySO> abilities = new List<AbstractAbilitySO>();
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/CreatureScripts/UnitAbilitiesContainer.cs:10]: Why was this changed? This seems to serve no purpose

File: Assets/CharacterFactory.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+            unit.transform.parent = unitHolder;
+
+            numUnits += 2;
+        }
+
+
+    }
+    
+}
```
Comment (wynn-rj) [inline Assets/CharacterFactory.cs]: This should create an object that extends an `ICharacter`, with a `GameObject` that wraps around that. There is no reason this should be implemented as a separate class. Everything from the database should be pulled into one of the data-based classes and then `GameObjects` should wrap those

**Review by wynn-rj — CHANGES_REQUESTED** [review]

**Comment by wynn-rj** [comment]
Overall the database side looks good, but I am very confused why the classes designed for interacting with the database were not used. If you have a good reason for this I am willing to accept this as long as an issue is created to fix this in the next timebox

File: Assets/CharacterFactory.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using SwordAndBored.Battlefield.CreaturScripts;
+using SwordAndBored.Battlefield;
+using Cinemachine;
+using SwordAndBored.Battlefield.CameraUtilities;
+using SwordAndBored.Battlefield.TurnMechanism;
+using SwordAndBored.GameData.Database;
+using SwordAndBored.GameData.Database.Tables;
+
```
Comment (wynn-rj) [inline Assets/CharacterFactory.cs]: No namespace is given. Also this should ***not*** be on the root folder Assets

File: Assets/Scripts/GameData/Database/UnitFromDatabase.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
+﻿using System.Data;
+using Mono.Data.Sqlite;
+using UnityEngine;
+using SwordAndBored.GameData.Database;
+using SwordAndBored.GameData.Database.Tables;
+using TMPro;
+
```
Comment (wynn-rj) [inline Assets/Scripts/GameData/Database/UnitFromDatabase.cs]: No namespace

**Review by wynn-rj — CHANGES_REQUESTED** [review]
An additional this is all of the table class implement the same functionality, so they should all extend a shared interface

File: Assets/Scripts/Battlefield/CreatureScripts/UnitStats.cs (hunk 1 above)
Comment (wynn-rj) [reply Assets/Scripts/Battlefield/CreatureScripts/UnitStats.cs:19]: Will be fixed in #93

File: Assets/CharacterFactory.cs (hunk 5 above)
Comment (wynn-rj) [reply Assets/CharacterFactory.cs]: Will be fixed in #93

**Review by ThBrokenToaster — APPROVED** [review]
*Thumbs Up* Good character spawning

**Review by wynn-rj — APPROVED** [review]

**Pull request merged**